lib/posix/init_clock_sntp.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(posix, CONFIG_POSIX_LOG_LEVEL);
+
+#include <errno.h>
+#include <init.h>
+#include <net/sntp.h>
+
+static int posix_init_clock_via_sntp(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	struct sntp_time ts;
+	struct timespec tspec;
+	int res = sntp_simple(CONFIG_POSIX_CLOCK_SNTP_INIT_SERVER,
```

Comment:
This will work only after networking is up, the IP address is set etc. So we might need here some more checks etc and possibly postpone this a bit after the net is up and running. To be investigated later when/if needed.

Reply:
> This will work only after networking is up, the IP address is set etc.

Yes, this is handled by carefully selecting default init priority to be after network auto-config, and mentioning this fact in Kconfig help.

One day, we might sit and review set of init priorities and see if we can make it "easier". (E.g. now, spacing is pretty tight between some levels.)

Reply:
The issue will be seen if you have DHCP enabled, I am pretty sure this time setting will fail in that case.